programs/test/benchmark.c

```diff
@@ -684,7 +685,8 @@ int main( int argc, char *argv[] )
     {
         mbedtls_des_context des;
         mbedtls_des_init( &des );
-        mbedtls_des_setkey_enc( &des, tmp );
+        if( mbedtls_des_setkey_enc( &des, tmp ) != 0 )
+            mbedtls_exit( 1 );
```

Comment:
I think this is a memleak for &des. Are we ok with it?

Reply:
I wonder why wasn't this caught by CI

Reply:
Is this a memory leak? Yes and no. The program exits, and that takes care of freeing all of its memory. We do want our test programs to explicitly free heap memory because we build and run them with memleak detection in some configurations (Asan, Valgrind) and we don't want to have false positives due to program global data in addition to potential true positives from inside the library. But this is an error case which is not exercised.

Furthermore, unless `MBEDTLS_DES_ALT` is active, a DES context doesn't have any embedded objects, so `mbedtls_des_free` doesn't actually free anything. And unless `MBEDTLS_DES_ALT` is active, `mbedtls_des_setkey_enc` can't fail, anyway.

In addition to potentially freeing memory, `mbedtls_xxx_free` also wipes confidential data, which we'd want to do at exit time in a production program. This doesn't matter in a benchmark program.

The benchmark program just exits without cleanup in most error conditions. Changing that would be out of scope here and not very useful.

Reply:
Well, of course the allocated memory is freed when app exits (OS takes care of that). So, even if there is no memleak in current usage, it might become a memleak if DES implementation changes.

Reply:
In principle, it would be cleaner to ensure `des_free()` is always called before exiting, but considering this is the benchmark program (which we don't really want to run under sanitizers), currently DES does not allocate memory, and is unlikely to change before it's removed, I agree with Gilles that just exiting without cleanup is acceptable here.

Reply:
Not cleaning up on errors is what the benchmark program already does. So in any case, if we want to do cleanup, that's out of scope of this pull request.

Reply:
ok